### ISSUE TITLE
Compass tutorial not working with Kyma 1.13

### ISSUE THREAD

**Issue by DimitarPetrov** [kyma-project/kyma#8733]
## Description
While executing the [compass tutorial](https://kyma-project.io/docs/1.13/components/compass/#tutorials-tutorials) I have encountered a few issues.
1. The order-service.yml is not adapted to work with the new v2 API Gateway and use the deprecated *API* custom resource instead of new *APIRule*.

2. When registering the application in compass and after that trying to provision an instance of that application via Service Catalog, the instance is stuck in provisioning state if I do not provide credentials in the compass when creating the API Package. This is not the case when registering the API Package with some dummy basic credentials.

3. After exposing the lambda it was not accessible from local machine.

4. The lambda function does not return any response after completing all the steps. The error is in the gateway:
```
time="2020-06-08T12:20:22Z" level=info msg="Modified request url : 'https://test.kyma.local/orders', schema : 'https', path : '/orders'"
2020/06/08 12:20:22 http: proxy error: dial tcp: lookup test.kyma.local on 10.96.0.10:53: no such host
```

## Proposed solutions/worarounds
1. The API resource in `order-service.yml` should be changed to APIRule. The version which worked for me is:
```
apiVersion: gateway.kyma-project.io/v1alpha1
kind: APIRule
metadata:
  name: order-service
spec:
  rules:
    - path: /.*
      methods: ["GET", "POST", "PUT", "DELETE"]
      accessStrategies:
        - handler: noop
  gateway: kyma-gateway.kyma-system.svc.cluster.local
  service:
    name: order-service
    port: 8017
    host: test
```

2. As a workaround I registered the API Package with dummy basic credentials despite of not secured order-service app. However this is a workaround and needs further investigation.

3. After exposing the lambda the host provided in the APIRule with the domain of the cluster should be added to local `/etc/hosts` file in order to access it from the local machine.

4. The url `test.kyma.local` which must be provided to compass when registering the application seems to be handled via Istio gateway and seems to be for external traffic only as I see from the logs.
 This is probably working for the multi tenant mode where the compass is outside of Kyma, but for the single tenant I managed to complete the tutorial with the FQN of the ClusterIP service of order-service. In my case the url was: `http://order-service.<namespace-name>.svc.cluster.local:8017 `. 
With this URL used for TargetURL when registering the application in compass the lambda works like expected.

**Comment by PK85** [comment]
Hello @DimitarPetrov,

good findings. My comments:
1) Yes we shoudl use APIRules. With APIRules we can:
  -  set OAuth2 auth and could be then configured in the Compass for given Application as defaultAuth
  - leave it as now unprotected and then configured in the Compass for given Application like documented here: https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-credential-requests.md#registering-package-api-without-credentials 

2) Unfortunately  Compass UI when setting Credentials to NONE is sets defaultCredentials to NONE. That means Application needs to provide credentials to Compass per every usage. We need to improve UI to allow to set defaultCredentisals#Credentials to null and then it will have default unprotected access, see: https://kyma-project.io/docs/1.13/components/compass#tutorials-manage-your-applications-using-kyma-console-and-compass-u-is

3) and 4) as I checked with team we tested that not locally. Tutorial shoudl include both local and cluster installation.
